pywbemtools/pywbemcli/_connection_repository.py

```diff
@@ -255,17 +287,37 @@ def _load_connections_file(self):
                        format(B08_DEFAULT_CONNECTIONS_FILE,
                               DEFAULT_CONNECTIONS_FILE))
 
+        # load the existing file
         if os.path.isfile(self._connections_file):
             with self._open_file(self._connections_file, 'r') as _fp:
                 try:
                     dict_ = yaml.safe_load(_fp)
-                    # put all the connection definitions into a group
-                    # in the connection file
-                    connections_dict = dict_[
-                        ConnectionRepository.connections_group_name]
 
-                    self._default_connection_name = dict_[
-                        ConnectionRepository.default_connection_grp_name]
+                    # Try building dictionary of server definitions
+                    try:
+                        connections_dict = dict_[
+                            ConnectionRepository.connections_group_name]
+                    except KeyError:
```

Comment:
1. In the docstring of this method, the documented exceptions do not match the actually raised exceptions.
2. If exceptions are raised in an except clause, set their `__cause__` attribute to `None` to avoid the re-raised messages.
3. See my comment above on a proper exception hierarchy.

Reply:
**Update:** I'll improve the exception handling of `ConnectionRepository` after this PR is merged, so no need to invest in that as part of this PR. You can ignore my comment above.

Reply:
I created an exception hierarchy.